core/src/test/java/org/apache/calcite/sql/test/SqlPrettyWriterTest.java

```diff
@@ -365,6 +365,14 @@ private SqlPrettyWriterFixture tableDotStar() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6102">[CALCITE-6102]
+   * SqlWriter in SqlInsert's unparse start a list but does not end it</a>. */
+  @Test void testInsert() {
+    sql("insert into t1 select * from t2")
```

Comment:
Can it reproduce the exception?
`Frame does not match current frame`

> Effective test cases should fail after patch removal.

Reply:
@JiajunBernoulli ok, i'll add a new test case which can throw this exception after patch removal